### PR TITLE
Magazines for Crew Served Weapons

### PR DESCRIPTION
**Pull request by PabstMirror**
Ref #4573

Adds unit carry-able magazine that can be loaded into static weapons.
Ideas borrowed partialy from ACE2.

Right now it just supports .50 BMG, .50 Soviet, A3 20mm GMG, Titans and the mk6 mortars (using the shells from mk6 ammo handling)

I would love to add belt linking like ace_reload, but most a3 weapons start with multiple magazines, which causes problems with `setVehicleAmmo`/`setMagazineTurretAmmo` commands.
Still trying to think of ways around this.

### CODE REVIEW

File: addons/crewserved/CfgMagazineGroups.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    
+    // Soviet HMG (12.7x108mm)
+    class ACE_50Rnd_127x108_carryable {
+        rhs_mag_127x108mm_50 = 1;
+        rhs_mag_127x108mm_100 = 1;
+        rhs_mag_127x108mm_150 = 1;
+        CUP_150Rnd_127x108_KORD_M = 1;
+        CUP_50Rnd_127x108_KORD_M = 1;
+        CUP_50Rnd_TE3_LRT5_127x107_DSHKM_M = 1; // not sure why cup uses 107 for the DSHKM?
+        CUP_150Rnd_TE3_LRT5_127x107_DSHKM_M = 1;
```
Comment (VKing6) [inline addons/crewserved/CfgMagazineGroups.hpp]: I'm pretty sure this was a typo introduced by BIS in Arma 2, or maybe even Arma 1.

File: addons/crewserved/CfgMagazineGroups.hpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        8Rnd_82mm_Mo_Flare_white = 1;
+    };
+    class ACE_1Rnd_82mm_Mo_HE_Guided {
+        8Rnd_82mm_Mo_guided = 1;
+    };
+    class ACE_1Rnd_82mm_Mo_HE_LaserGuided {
+        8Rnd_82mm_Mo_LG = 1;
+    };
+    
+    // A3 Titans (Spike) - just use handheld magazines (Does this make sense?)
```
Comment (VKing6) [inline addons/crewserved/CfgMagazineGroups.hpp]: Yea, It's supposed to be the same launcher on a tripod, so popping the same missile in either makes sense.

File: addons/crewserved/functions/fnc_actionsUnload.sqf [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+private _actions = [];
+private _handeledMagTypes = [];
+
+private _statement = {
+    params ["_target", "_player", "_params"];
+    _params params ["_vehMag", "_turretPath", "_carryMag"];
+    TRACE_5("Starting unload mag progres bar",_target,_turretPath,_player,_carryMag,_vehMag);
+
+    if (!(_player canAdd _carryMag)) exitWith {
```
Comment (VKing6) [inline addons/crewserved/functions/fnc_actionsUnload.sqf]: Drop magazine on the ground if there's no room in the inventory?

**Comment by VKing6** [comment]
Intended that you can load magazines into a weapon with magazines in it?

**Comment by PabstMirror** [comment]
@VKing6 - There is an ace setting for that.

Most guns come with multiple mags in them (e.g. the b_hmg comes with 5x `100Rnd_127x99_mag_Tracer_Red`)
I wasn't sure the best way to handle this.  After shooting 3 belts on a static weapon, one would think they could re-add belts to return it to it's original state.

Ideally we could guarantee all statics just come with a single magazine loaded, but I don't think that's possible.

**Comment by VKing6** [comment]
Ah, yes, I see how the module setting works now.

Yea, that'll probably work okay.

**Comment by Whipgit** [comment]
Is this still a WIP?

**Comment by PabstMirror** [comment]
Merged into #5652

**Pull request closed without merging**